File: backend/src/main/java/com/gxs/bank/repository/KycDocumentRepository.java
package com.gxs.bank.repository;

import com.gxs.bank.model.KycDocument;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import java.util.List;
import java.util.UUID;

@Repository
public interface KycDocumentRepository extends JpaRepository<KycDocument, UUID> {
    List<KycDocument> findByUserId(UUID userId);
    List<KycDocument> findByVerificationStatus(KycDocument.VerificationStatus status);
}

File: backend-python/com/gxs/bank/repository/KycDocumentRepository.py
from com.gxs.bank.model.KycDocument import KycDocument
from com.gxs.bank.repository._base import BaseRepository


class KycDocumentRepository(BaseRepository):
    def __init__(self, db):
        super().__init__(db, KycDocument)

    def findByUserId(self, userId: str):
        return self.db.query(KycDocument).filter(KycDocument.userId == userId).all()

    def findByVerificationStatus(self, status):
        return self.db.query(KycDocument).filter(KycDocument.verificationStatus == status).all()
